Mathlib/MeasureTheory/Measure/HasOuterApproxClosed.lean

```diff
@@ -0,0 +1,239 @@
+
+lemma indicator_le_apprSeq (n : ℕ) :
+    indicator F (fun _ ↦ 1) ≤ hF.apprSeq n := by
+  intro x
+  by_cases hxF : x ∈ F
+  · simp only [hxF, indicator_of_mem, one_le_apprSeq_apply hF n]
+  · simp only [hxF, not_false_eq_true, indicator_of_not_mem, zero_le]
+
+lemma apprSeq_apply_eq_one (n : ℕ) {x : X} (hx : x ∈ F) :
+    hF.apprSeq n x = 1 :=
+  le_antisymm (apprSeq_apply_le_one hF n x) (one_le_apprSeq_apply hF n hx)
+
+/-- The measure of a closed set is at most the integral of any function in a decreasing
+approximating sequence to the indicator of the set. -/
+theorem measure_le_lintegral [MeasurableSpace X] [OpensMeasurableSpace X] (μ : Measure X) (n : ℕ) :
+    μ F ≤ ∫⁻ x, (hF.apprSeq n x : ℝ≥0∞) ∂μ := by
+  convert_to ∫⁻ x, (F.indicator (fun _ ↦ (1 : ℝ≥0∞))) x ∂μ ≤ ∫⁻ x, hF.apprSeq n x ∂μ
+  · rw [lintegral_indicator _ hF.measurableSet]
+    simp only [lintegral_one, MeasurableSet.univ, Measure.restrict_apply, univ_inter]
+  · apply lintegral_mono
+    intro x
+    by_cases hxF : x ∈ F
+    · simpa only [hxF, indicator_of_mem, one_le_coe_iff] using one_le_apprSeq_apply hF n hxF
+    · simp only [hxF, not_false_eq_true, indicator_of_not_mem, zero_le]
+
+/-- The integrals along a decreasing approximating sequence to the indicator of a closed set
+tend to the measure of the closed set. -/
+lemma tendsto_lintegral_apprSeq [MeasurableSpace X] [OpensMeasurableSpace X]
+    (μ : Measure X) [IsFiniteMeasure μ] :
+    Tendsto (fun n ↦ ∫⁻ x, hF.apprSeq n x ∂μ) atTop (𝓝 ((μ : Measure X) F)) :=
+  measure_of_cont_bdd_of_tendsto_indicator μ hF.measurableSet hF.apprSeq
+    (apprSeq_apply_le_one hF) (tendsto_apprSeq hF)
+
+end HasOuterApproxClosed --namespace
+
+noncomputable instance (X : Type _) [TopologicalSpace X]
+    [TopologicalSpace.PseudoMetrizableSpace X] : HasOuterApproxClosed X := by
+  letI : PseudoMetricSpace X := TopologicalSpace.pseudoMetrizableSpacePseudoMetric X
+  refine ⟨fun F hF ↦ ?_⟩
+  · use fun n ↦ thickenedIndicator (δ := (1 : ℝ) / (n + 1)) Nat.one_div_pos_of_nat F
+    refine ⟨?_, ⟨?_, ?_⟩⟩
+    · exact fun n x ↦ thickenedIndicator_le_one Nat.one_div_pos_of_nat F x
+    · exact fun n x hxF ↦ one_le_thickenedIndicator_apply X Nat.one_div_pos_of_nat hxF
+    · have key := thickenedIndicator_tendsto_indicator_closure
+                (δseq := fun (n : ℕ) ↦ (1 : ℝ) / (n + 1))
+                (fun _ ↦ Nat.one_div_pos_of_nat) tendsto_one_div_add_atTop_nhds_0_nat F
+      rw [tendsto_pi_nhds] at *
+      intro x
+      nth_rw 2 [←IsClosed.closure_eq hF]
+      exact key x
+
+namespace MeasureTheory
+
+/-- Two finite measures give equal values to all closed sets if the integrals of all bounded
+continuous functions with respect to the two measures agree. -/
+theorem measure_isClosed_eq_of_forall_lintegral_eq_of_isFiniteMeasure {Ω : Type*}
+    [MeasurableSpace Ω] [TopologicalSpace Ω] [HasOuterApproxClosed Ω]
+    [OpensMeasurableSpace Ω] {μ ν : Measure Ω} [IsFiniteMeasure μ] [IsFiniteMeasure ν]
```

Comment:
You don't need `IsFiniteMeasure`, as you can deduce it from your assumption applied to the constant function 1. Same thing in next lemma.